provider/core/src/response.rs

```diff
@@ -72,7 +73,45 @@ pub struct DataPayload<M>
 where
     M: DataMarker,
 {
-    pub(crate) yoke: Yoke<M::Yokeable, Option<RcWrap<[u8]>>>,
+    pub(crate) yoke: Yoke<M::Yokeable, Option<Cart>>,
+}
+
+/// The type of the "cart" that is used by `DataPayload`.
+#[derive(Clone)]
+pub struct Cart(
+    #[cfg(feature = "sync")] yoke::erased::ErasedArcCart,
+    #[cfg(not(feature = "sync"))] yoke::erased::ErasedRcCart,
+);
+
+impl Deref for Cart {
+    type Target = dyn yoke::erased::ErasedDestructor;
+    fn deref(&self) -> &Self::Target {
+        &*self.0
+    }
+}
+// Safe because both Rc and Arc are CloneableCart
+unsafe impl stable_deref_trait::StableDeref for Cart {}
+// Safe because both Rc and Arc are StableDeref
+unsafe impl yoke::CloneableCart for Cart {}
+
+impl Cart {
+    /// Creates a Yoke<Y, Option<Cart>> from some owned C by applying f.
+    pub fn make_yoke<Y, F, E, C, D>(cart: C, f: F) -> Result<Yoke<Y, Option<Self>>, E>
+    where
+        for<'a> Y: Yokeable<'a>,
+        F: FnOnce(&D) -> Result<<Y as Yokeable>::Output, E>,
+        C: Deref<Target = D> + crate::MaybeSendSync + 'static,
+        D: 'static + ?Sized,
+    {
+        // Safe because the cart is only wrapped
+        unsafe {
+            Ok(
+                Yoke::try_attach_to_cart(SelectedRc::new(cart), |b: &C| f(&*b))?
+                    .replace_cart(|rc| Cart(rc))
+                    .wrap_cart_in_option(),
+            )
+        }
```

Comment:
Nit (style): Write it such that only the unsafe function is inside the unsafe

```suggestion
        Yoke::try_attach_to_cart(SelectedRc::new(cart), |b: &C| f(&*b))
            // Safe because the cart is only wrapped
            .map(|yoke| unsafe { yoke.replace_cart(|rc| Cart(rc)) })
            .wrap_cart_in_option()
```